packages/insured-bridge-relayer/src/CrossDomainFinalizer.ts

```diff
@@ -0,0 +1,91 @@
+import winston from "winston";
+
+import { createEtherscanLinkMarkdown, createFormatFunction, PublicNetworks } from "@uma/common";
+import { getAbi } from "@uma/contracts-node";
+import { InsuredBridgeL1Client, InsuredBridgeL2Client, GasEstimator } from "@uma/financial-templates-lib";
+
+export enum RelaySubmitType {
+  Slow,
+  SpeedUp,
+  Instant,
+  Ignore,
+}
+
+export class CrossDomainFinalizer {
+  constructor(
+    readonly logger: winston.Logger,
+    readonly gasEstimator: GasEstimator,
+    readonly l1Client: InsuredBridgeL1Client,
+    readonly l2Client: InsuredBridgeL2Client,
+    readonly account: string
+  ) {}
+  async checkForBridgeableL2TokensAndBridge() {
+    this.logger.debug({ at: "AcrossRelayer#CrossDomainFinalizer", message: "Checking bridgeable L2 tokens" });
+
+    const whitelistedTokenEvents = await this.l1Client.bridgeAdmin.getPastEvents("WhitelistToken", {
+      fromBlock: 0,
+      toBlock: await this.l1Client.l1Web3.eth.getBlockNumber(),
```

Comment:
We should ideally parameterize the `fromBlock` as `this.firstBlockToSearch` and `toBlock` as `this.endingBlockNumber` so that on each subsequent run, this bot only looks up new events.

Perhaps we should build a `BridgeAdminClient` that handles event processing....

Reply:
we _could_....is it worth it though for this one look back? also this is running on L1 and infura has no problem in calling this kind of method.

we could also store the deployment block of the bridge admin in a config file to decrease how far back this has to look?

WDYT

Reply:
Unrelated to the discussion above: why do we need a toBlock at all?

Reply:
@mrice32 good point! removing this and leaving it just with the `fromBLock` works just fine.